src/optim/adam/adam.cu

```diff
@@ -56,11 +58,62 @@ __device__ void adam_update(
     param[i] -= g;
 }
 
+__device__ void adam_update(
```

Comment:
Can remove this since we aren't doing this for Cpu. Pytorch also has accuracy issues if doing training in pure f16